lib/block_view_message.go

```diff
@@ -825,6 +856,76 @@ func (bav *UtxoView) _connectMessagingGroup(
 		}
 	}
 
+	// V3 Messages: MUTING and UNMUTING members
+	if blockHeight >= bav.Params.ForkHeights.DeSoV3MessagesMutingBlockHeight {
+		if existingEntry != nil && !existingEntry.isDeleted {
+			value, operationTypeExists := txn.ExtraData["OperationType"]
+			if operationTypeExists {
+				// inline helper method
+				contains := func(memberList []*MessagingGroupMember, member *MessagingGroupMember) bool {
+					for _, a := range memberList {
+						if a == member {
+							return true
+						}
+					}
+					return false
+				}
+				// make deep copy of existingEntry.MuteList to prevent mempool errors
+				muteListCopy := make([]*MessagingGroupMember, len(existingEntry.MuteList))
+				copy(muteListCopy, existingEntry.MuteList)
+				// check if mute or unmute
+				if string(value) == MessagingGroupOperationMute {
+					for _, s := range txMeta.MessagingGroupMembers {
+						// Add s to muteList
+						// Make sure does not already exist to ensure no dups
+						if !contains(muteListCopy, s) {
+							muteListCopy = append(muteListCopy, s)
+						}
+					}
+				} else if string(value) == MessagingGroupOperationUnmute {
+					for _, s := range txMeta.MessagingGroupMembers {
+						for i, toUnmute := range muteListCopy {
+							if reflect.DeepEqual(toUnmute.GroupMemberPublicKey[:], s.GroupMemberPublicKey[:]) {
+								muteListCopy = append(muteListCopy[:i], muteListCopy[i+1:]...)
+								break
+							}
+						}
+					}
+				}
+				// Finally set the real existingEntry.MuteList to point to the clone
+				existingEntry.MuteList = muteListCopy
+				// Create a MessagingGroupEntry so we can add the entry to UtxoView.
+				messagingGroupEntry := MessagingGroupEntry{
+					GroupOwnerPublicKey:   &messagingGroupKey.OwnerPublicKey,
+					MessagingPublicKey:    messagingPublicKey,
+					MessagingGroupKeyName: NewGroupKeyName(txMeta.MessagingGroupKeyName),
+					MessagingGroupMembers: existingEntry.MessagingGroupMembers,
+					MuteList:              existingEntry.MuteList,
+					ExtraData:             existingEntry.ExtraData,
+				}
+				// Create a utxoOps entry, we make a copy of the existing entry.
+				var prevMessagingKeyEntry *MessagingGroupEntry
+				if existingEntry != nil && !existingEntry.isDeleted {
```

Comment:
I think we already checked for this condition?

Reply:
yes, fixed.